tests/executor/exchange_seq_scan_test.cpp

```diff
@@ -0,0 +1,346 @@
+//
+// Created by wendongli on 4/17/16.
+//
+
+#include <chrono>
+#include <ctime>
+#include <memory>
+#include <set>
+#include <string>
+#include <vector>
+
+#include "harness.h"
+
+#include "backend/catalog/schema.h"
+#include "backend/common/types.h"
+#include "backend/common/value.h"
+#include "backend/common/value_factory.h"
+#include "backend/concurrency/transaction.h"
+#include "backend/concurrency/transaction_manager_factory.h"
+#include "backend/executor/abstract_executor.h"
+#include "backend/executor/exchange_seq_scan_executor.h"
+#include "backend/executor/executor_context.h"
+#include "backend/executor/logical_tile.h"
+#include "backend/executor/logical_tile_factory.h"
+#include "backend/executor/seq_scan_executor.h"
+#include "backend/expression/abstract_expression.h"
+#include "backend/expression/expression_util.h"
+#include "backend/planner/seq_scan_plan.h"
+#include "backend/storage/data_table.h"
+#include "backend/storage/tile_group_factory.h"
+
+#include "executor/executor_tests_util.h"
+#include "executor/mock_executor.h"
+#include "harness.h"
+
+using ::testing::NotNull;
+using ::testing::Return;
+
+namespace peloton {
+namespace test {
+
+class ExchangeSeqScanTests : public PelotonTest {};
+const std::set<oid_t> g_tuple_ids({0, 3});
+
+/*
+ * create a table with tile_group_num tiles, each of which has row_num rows.
+ * heterogeneous table.
+ */
+storage::DataTable *CreateTable(size_t tile_group_num, size_t row_num) {
+  std::unique_ptr<storage::DataTable> table(
+      ExecutorTestsUtil::CreateTable(row_num, false));
+  TestingHarness::GetInstance().GetNextTileGroupId();
+  size_t index = 0;
+  ExecutorTestsUtil::PopulateTiles(table->GetTileGroup(index++), row_num);
+  if (tile_group_num == 1) {
+    return table.release();
+  }
+  if (tile_group_num % 2 == 0) {
```

Comment:
This changes the behavior depending on if the number of tile groups is even. This is not made clear to the caller of this function and should be made explicit